fili-core/src/main/java/com/yahoo/bard/webservice/data/filterbuilders/ConjunctionDruidFilterBuilder.java

```diff
@@ -30,7 +35,15 @@
  * each dimension being filtered on.
  */
 public abstract class ConjunctionDruidFilterBuilder implements DruidFilterBuilder {
+
     private static final Logger LOG = LoggerFactory.getLogger(ConjunctionDruidFilterBuilder.class);
+    private static final SystemConfig SYSTEM_CONFIG = SystemConfigProvider.getInstance();
+
+    private static final int DEFAULT_MAX_NUM_DRUID_FILTERS = 10;
```

Comment:
> High enough that you'll start experiencing some pain before
you hit the ceiling.

This can be dangerous because the pain is likely felt across multiple requests as a generally elevated latency. I think it's OK to have the default limit start _very high_ (eg. 100k) when the feature is introduced, since that should generally be backwards-compatible, but I think the default limit should be lowered for _new_ installations after some sane period of time.

What I'm essentially saying is that we should deprecate the high limit and lower it in a later version. That gives folks time to adjust the config value as needed, while also retaining a general policy of "sane by default".

I agree than the "safe" limit needs to be much higher than 10, but 100k seems a bit too high, since that would generate a massive query that would incur ser/deser and over-the-wire latency. 1k seems reasonable, but I could be convinced 10k is sane if there were some "here's the impact that size has on latency" data.

Reply:
@cdeszaq I have an issue created for your request - https://github.com/yahoo/fili/issues/705